lib/tasks/bundlers/generateStandaloneAppBundle.js

```diff
@@ -101,6 +101,35 @@ module.exports = async function({workspace, dependencies, taskUtil, options: {pr
 		filters = ["jquery.sap.global.js"];
 	}
 
+	const unoptimizedModuleNameMapping = {};
+	let unoptimizedResources = resources;
+	if (taskUtil) {
+		unoptimizedResources = await new ReaderCollectionPrioritized({
+			name: `generateStandaloneAppBundle - prioritize workspace over dependencies: ${projectName}`,
+			readers: [workspace, dependencies]
+		}).filter(function(resource) {
+			// Remove any non-debug variants
+			return !taskUtil.getTag(resource, taskUtil.STANDARD_TAGS.HasDebugVariant);
+		}).byGlob("/resources/**/*.{js,json,xml,html,properties,library,js.map}");
+
+		// For "unoptimized" bundles, the non-debug files have already been filtered out above.
+		// Now we need to create a mapping from the debug-variant resource path to the respective module name,
+		// which is basically the non-debug resource path, minus the "/resources/"" prefix.
+		// This mapping overwrites internal logic of the LocatorResourcePool which would otherwise determine
+		// the module name from the resource path, which would contain "-dbg" in this case. That would be
+		// incorrect since debug-variants should still keep the original module name.
+		for (let i = unoptimizedResources.length - 1; i >= 0; i--) {
+			const resourcePath = unoptimizedResources[i].getPath();
+			if (taskUtil.getTag(resourcePath, taskUtil.STANDARD_TAGS.IsDebugVariant)) {
+				const nonDbgPath = ModuleName.getNonDebugName(resourcePath);
+				if (!nonDbgPath) {
+					throw new Error(`Failed to resolve non-debug name for ${resourcePath}`);
+				}
+				unoptimizedModuleNameMapping[resourcePath] = nonDbgPath.slice("/resources/".length);
+			}
+		}
```

Comment:
We have this code in three places now, so it would be better to refactor it

Reply:
As discussed, maybe we could have all bundling tasks call the `generateBundle`-task  or create a "helper" module in `lib/tasks/bundlers/` for this purpose 👍